### DISCUSSION TITLE
[adapter-vercel] Deploy Sveltekit to the edge but keep API routes as serverless functions?

### DISCUSSION THREAD

**Discussion by madeleineostoja** [sveltejs/kit#8310] [Q&A] [answered]
I have sveltekit apps I'd like to deploy to Vercel edge functions using the new `edge` option in `adapter-vercel`. All the app's routes themselves should work just fine on the edge, but I also have several API endpoints (ie: `+server.ts` routes) that make use of Node APIs, which won't work as edge functions. All my API routes live in one place (`/api`), and I'm wondering if there's a way to be more selective with what gets pushed to the edge and what gets deployed as serverless functions. 

I don't need the performance of the edge for those endpoints, and trying to make all of them (and their dependencies) run in the very limited environment of edge functions would be a maintenance nightmare. 

Is this something that Sveltekit could detect somehow? Or could we pass a route pattern of serverless vs edge functions? Or even just a simple boolean to deploy `+server` routes as serverless functions? I imagine most `+server` routes would rely on Node APIs to some extent.

**Comment by dummdidumm** [comment, marked as answer]
There's no possibility of this at the moment, but it's on our radar and there's a feature request about it somewhere

**Reply by madeleineostoja** [reply]
Awesome! I couldn’t find a ticket for it but I’ll look again. I’d love to track progress on it as an enhancement I can implement when it ships

**Reply by teemingc** [reply]
I believe the idea was mentioned here https://github.com/sveltejs/kit/discussions/7716 . We're already one step closer!

**Reply by Rich-Harris** [reply]
I opened an issue for it just now: https://github.com/sveltejs/kit/issues/8383

**Reply by madeleineostoja** [reply]
Thanks Rich! The solution outlined in that issue sounds like a perfect fit
